### DISCUSSION TITLE
Is there a way to render a radio group that looks like attached IconButtons?

### DISCUSSION THREAD

**Discussion by AndrewPrifer** [chakra-ui/chakra-ui#2385] [Q&A] [answered]
Hi, I am trying to make radio buttons out of attached buttons. I may be approaching the implementation the wrong way, however currently, it works when I use buttons (slightly modified version of the example in docs):

```ts
interface RadioCardProps extends UseRadioProps {
  label: string;
  icon: ReactElement<IconType>;
}

const RadioCard: VFC<RadioCardProps> = (props) => {
  const { getInputProps, getCheckboxProps } = useRadio(props);

  const input = getInputProps();
  const checkbox = getCheckboxProps();

  return (
    <>
      <Button
        leftIcon={props.icon}
        as="label"
        {...checkbox}
        _checked={{
          bg: 'teal.600',
          color: 'white',
          borderColor: 'teal.600',
        }}
      >
        {props.label}
        <input {...input} />
      </Button>
    </>
  );
};

export interface TransformControlModeRadioProps {
  value: TransformControlMode;
  onChange: (value: TransformControlMode) => void;
}

const TransformControlModeRadio: VFC<TransformControlModeRadioProps> = ({
  value,
  onChange,
}) => {
  const { getRootProps, getRadioProps } = useRadioGroup({
    name: 'transformControlsMode',
    value,
    onChange,
  });

  const group = getRootProps();

  return (
    <ButtonGroup {...group} isAttached>
      <RadioCard
        {...getRadioProps({ value: 'translate' })}
        label="Translate"
        icon={<GiMove />}
      />
      <RadioCard
        {...getRadioProps({ value: 'rotate' })}
        label="Rotate"
        icon={<GiClockwiseRotation />}
      />
      <RadioCard
        {...getRadioProps({ value: 'scale' })}
        label="Scale"
        icon={<GiResize />}
      />
    </ButtonGroup>
  );
};
```

However if I try to use IconButtons, it stops working because IconButtons don't render their children prop (even though Typescript allows me to pass one, which is necessary to render the hidden input element.

Is there a way to render a radio group that looks like attached IconButtons? Or should I approach this differently? Looking at the documentation, it seems I can only attach buttons to each other.

**Comment by with-heart** [comment]
Hi @AndrewPrifer! I'd love to take a look for you. Any way you could make a working CodeSandbox of what you're trying to accomplish to make that easier?

**Reply by AndrewPrifer** [reply]
Hey @with-heart , this codesandbox contains both the working Button implementation and the not working IconButton one: https://codesandbox.io/s/recursing-frog-4yxmt?file=/src/index.tsx

Thank you for your help!

**Comment by with-heart** [comment, marked as answer]
@AndrewPrifer I built this off of your CodeSandbox example: https://codesandbox.io/s/nervous-bohr-wntmb?file=/src/TransformControlModeIcons.tsx

It renders `input` next to the `label` instead of as a child, and associates to it using `htmlFor={input.id}`.

**Reply by AndrewPrifer** [reply]
Oh goddamnit, I tried this exact trick at one point, I just forgot you had to use htmlFor in React 🤦‍♂️ Thanks so much!

**Reply by with-heart** [reply]
Sometimes it's the smallest things that get in the way 🤣
